homeassistant/components/livisi/coordinator.py

```diff
@@ -0,0 +1,139 @@
+"""Code to manage fetching LIVISI data API."""
+from __future__ import annotations
+
+from datetime import timedelta
+from typing import Any
+
+from aiolivisi import AioLivisi, LivisiEvent, Websocket
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import (
+    AVATAR_PORT,
+    CLASSIC_PORT,
+    DEVICE_POLLING_DELAY,
+    LIVISI_DISCOVERY_NEW,
+    LOGGER,
+    PSS_DEVICE_TYPE,
+)
+
+
+class LivisiDataUpdateCoordinator(DataUpdateCoordinator[dict[str, Any]]):
+    """Class to manage fetching LIVISI data API."""
+
+    def __init__(
+        self, hass: HomeAssistant, config_entry: ConfigEntry, aiolivisi: AioLivisi
+    ) -> None:
+        """Initialize my coordinator."""
+        super().__init__(
+            hass,
+            LOGGER,
+            name="Livisi devices",
+            update_interval=timedelta(seconds=DEVICE_POLLING_DELAY),
+        )
+        self.config_entry: ConfigEntry = config_entry
+        self.hass: HomeAssistant = hass
+        self.aiolivisi: AioLivisi = aiolivisi
+        self.websocket: Websocket = Websocket(aiolivisi)
+        self.devices: list[dict[Any, Any]] = []
+        self.switch_devices: list = []
+        self.rooms: dict[str, Any] = {}
+        self.serial_number: str | None
+        self.controller_type: str | None
+        self.os_version: str | None
+        self.is_avatar: bool = False
+        self.port: str = ""
+
+    async def _async_update_data(self):
+        """Set device list."""
+        shc_devices: dict[Any, Any] = await self.async_get_devices()
+        for device in shc_devices:
+            if device in self.devices:
+                continue
+            if device["type"] == PSS_DEVICE_TYPE:
+                if device not in self.switch_devices:
+                    self.switch_devices.append(device)
+                    async_dispatcher_send(self.hass, LIVISI_DISCOVERY_NEW, device)
+            self.devices.append(device)
+
+    async def async_setup(self) -> None:
+        """Set up the Livisi Smart Home Controller."""
+        if bool(self.aiolivisi.livisi_connection_data) is False:
+            livisi_connection_data = {
+                "ip_address": self.config_entry.data["host"],
+                "password": self.config_entry.data["password"],
+            }
+
+            await self.aiolivisi.async_set_token(
+                livisi_connection_data=livisi_connection_data
+            )
+        controller_info = await self.aiolivisi.async_get_controller()
+
+        controller_data = controller_info
```

Comment:
Why are we using a different variable name than `controller_info`. `controller_data` is still pointing to the same object.

Reply:
I removed the unnecessary variable.